keps/sig-testing/2539-continuously-deploy-k8s-prow/README.md

```diff
@@ -0,0 +1,149 @@
+
+Historically, the biggest reasons why the manual processes were introduced are:
+
+- Errors in prow were not easily discoverable, most of time were reported by prow users, which was bad.
+
+This problem has been largely solved by the introduction of prow monitoring + alerting by grafana, prometheus, and prometheus alertmanager stack. And based on our experience in the past quarter, we haven’t had a single case where prow errors were discovered by humans earlier than by prow alerts. This fact gives us reasonable confidence to proceed with continuous delivery, and the following assumptions should hold:
+
+- Prow is stable as long as there is no alert. (Indicate that no need to inspect prow logs before bumping)
+- Errors caused by prow upgrades are discovered by prow alerts in a timely manner.
+
+
+### Goals
+
+The proposed change switches from daily manual deployment to hourly automated deployment.
+
+#### Prow Users
+
+Shouldn’t see any change, prow breakage should be discovered by prow monitoring system and rollback will be performed. The chance of prow being break is almost identical to what we have today(Assume there are not more than a single breaking change every day).
+
+#### Prow Oncall
+
+- What’s Not Changed
+  - React to prow alerts and take actions.
+- What’s Changed
+  - No more manual inspecting prow healthiness.
+  - No more manual lgtm/approve/retest autobump PRs.
+  - No more manual Slack posting.
+
+
+### Non-Goals
+
+Change how prow is released.
+
+
+## Proposal
+
+Prow autobump PRs are automatically merged every hour, only on working hours of working days.
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Breaking Changes in Prow
+Breaking changes in prow will require manual intervention. Currently prow isn’t able to handle these intelligently, as it was not designed with the mindset of API versions and thus kubernetes conversion webhook can not help coping with breaking changes among major APIs.
+One possible way of dealing with breaking changes, is:
+- Prow oncall inspects prow logs and breaking changes announcements once per week, and take actions based on deprecation warnings from prow logs and breaking changes from ANNOUNCEMENTS.md.
+- [Stretch Goal][Push left] Discover breaking changes, especially configs or flag changes in prow integration test (This requires prow integration test use the same set of deployment configs as prod)
+- [Optional] (This is not very reliable) Prow TLs inspects new PRs, manually identifies possible breaking changes and informs oncall for awareness. Either prow TL or oncall can take deeper look at the new PR and decide whether to take actions or  not.
+
+## Design Details
+
+#### Automated Merging of Prow Autobump PRs
+
+- Prow autobump job is already configured to run on work days only, change it to at least one hour apart, so that it doesn’t bump more frequently than one hour.
```

Comment:
We should decouple prow auto bump from job image auto bump (I have an open issue for this in test-infra)

Reply:
This is a requirement, IMO, since we're going to alter the frequency and they're currently coupled.

Reply:
https://github.com/kubernetes/test-infra/issues/21137 there's the issue for it